### ISSUE TITLE
Feature: Add passing pyglet encoders and/or automatic Ogg support

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#2049]
## Enhancement request:

TL;DR: @DigiDuncan pointed out we can't really load .ogg through arcade

### What should be added/changed?

1. Add a way to pass a decoder supported by pyglet's `media.load` function
2. (Optional / risky) Smart and helpful `pyogg` / `PyOggDecoder` warnings

### What would passing in a decoder would it help with?

Improve arcade user experience by allowing sound loading to:
1. take specific decoders such as pyglet's `PyOggDecoder`
2. avoid subclassing `Sound` as in [Charm's current code](https://github.com/DigiDuncan/Charm/blob/1b3f5a0ec8b822d914429a34bed869e1a370e4bf/charm/lib/oggsound.py)
3. without having to file tickets for new formats (example: [QOA](https://qoaformat.org/))

### Changes needed:

1. Add a `decoder` keyword argument to `arcade.sound.Sound.__init__`
2. ~~Add a temp `Decoder` protocol to annotate it before upstreaming matching [pyglet's codecs](https://github.com/pyglet/pyglet/tree/28e58b9e7f2d8e3d36b66ae24a25f5c5749ff3b6/pyglet/media/codecs)~~ Use [the existing `MediaDecoder` type](https://github.com/pyglet/pyglet/blob/28e58b9e7f2d8e3d36b66ae24a25f5c5749ff3b6/pyglet/media/codecs/__init__.py#L18)
3. Document the change
4. Update the sound page in the programming guide

### Follow-up Work

1. Add an example
2. Make upstream pyglet PRs to help with https://github.com/pyglet/pyglet/issues/1084:
   1. ~~Define a `Decoder` `Protocol` type matching pyglet's [pyglet's codecs](https://github.com/pyglet/pyglet/tree/28e58b9e7f2d8e3d36b66ae24a25f5c5749ff3b6/pyglet/media/codecs)~~
   2. Add argument and return annotations to the codecs

**Comment by DigiDuncan** [comment]
Thanks for writeup, pushfoo! +1.

**Comment by pushfoo** [comment]
Follow-up from earlier reading tonight:
1. There's already [a `MediaDecoder` base type in pyglet](https://github.com/pyglet/pyglet/blob/28e58b9e7f2d8e3d36b66ae24a25f5c5749ff3b6/pyglet/media/codecs/__init__.py#L18)
2. Although we could define a `MediaDecoderLike` protocol, it may be redundant for now
